### PR TITLE
Button: Focus in high contrast mode fix

### LINKED ISSUE #3419: Accessibility issue - buttons
### Bug Report

- __Package version(s)__:5.8.0
- __Browser and OS versions__: Edge

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: Normal

#### Describe the issue:

In high contrast, there is no way to distinguish a foucssed button from an unfocussed one.

Unfocussed - 
image

Focussed-
image

### PR DESCRIPTION
**Pull request by lynamemi**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #3419 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Button focus wasn't working in high contrast mode. To fix this, I refactored getFocusStyle to take a highContrastStyle object that defines focus style in high contrast mode. CommandBarButton has a different style than the rest.

I debated about putting more logic inside getFocusStyle, but landed on passing a generic IRawStyle object for more flexibility.

**In High Contrast White:**
image
image
image
image
image

#### Focus areas to test

I tested that focus still works in non high contrast.

### CODE REVIEW

File: packages/styling/src/styles/getFocusStyle.ts [hunk 1]
```diff
@@ -9,12 +10,14 @@ import { ITheme } from '../interfaces/index';
...
  * be 'relative' or 'absolute' so that the focus border can live around it.
+ * @param highContrastStyle - Style for high contrast mode.
  * @returns The style object.
  */
 export function getFocusStyle(
   theme: ITheme,
   inset: number = 0,
-  position: 'relative' | 'absolute' = 'relative'
+  position: 'relative' | 'absolute' = 'relative',
+  highContrastStyle: IRawStyle | undefined = undefined
```
Comment (phkuo) [inline packages/styling/src/styles/getFocusStyle.ts]: Is there a difference between this and just "IRawStyle?"?

Comment (lynamemi) [reply packages/styling/src/styles/getFocusStyle.ts]: Probably not different, but I am getting a tslint error if I don't have the extra bits when I `rush rebuild`:
`missing-optional-annotation  Argument following optional argument missing optional annotation: 
  highContrastStyle: IRawStyle`

**Comment by lynamemi** [comment]
bump for this PR. we're getting more issues reported for this bug.

File: packages/styling/src/styles/getFocusStyle.ts (hunk 1 above)
Comment (micahgodbolt) [reply packages/styling/src/styles/getFocusStyle.ts]: I can see why it needs to be IRawStyle or undefined, but do you need to set the default to undefined? isn't that what it'll be if it's not set?

Comment (lynamemi) [reply packages/styling/src/styles/getFocusStyle.ts]: Alas, that's part of what triggers the error I noted above.

**Comment by lynamemi** [comment]
@phkuo Any more feedback?

**Comment by yaananth** [comment]
High contrast black theme the background of the selected button (which should be similar to focus) should be in "Cyan" and button text should be in "Black" colour.

In white theme background colour should be in "Purple" and button text colour should be in "White".

Are we doing that here? From description, it doesn't look like it.

**Comment by lynamemi** [comment]
@yaananth, this PR is just for button focus in high contrast mode. I worked with our designer, @betrue-final-final. If it's another high contrast bug you're seeing, please file a separate issue.

**Comment by yaananth** [comment]
@lynamemi  That's where the confusion is, how do we distinguish focus vs selection for a button in web world? Is there a defintitive guidance here?

consider VS:
image

focusing, adopts "select" styles.
If we follow that, then focus should render "Select" styles for buttons in web.

**Comment by betrue-final-final** [comment]
Focus is a border and there is no selected button state. It’s either rest or hover or pressed.

**Comment by yaananth** [comment]
Thanks @betrue-final-final  as long as everyone across org is considering the same, sounds good!

**Pull request merged**